synapse/handlers/message.py

```diff
@@ -84,14 +84,22 @@ def __init__(self, hs):
                 "_schedule_next_expiry", self._schedule_next_expiry
             )
 
-    @defer.inlineCallbacks
-    def get_room_data(
-        self, user_id=None, room_id=None, event_type=None, state_key="", is_guest=False
-    ):
+    async def get_room_data(
+        self,
+        user_id: Optional[str] = None,
+        room_id: Optional[str] = None,
```

Comment:
Hmm, I know the previous docstring had it, but I'm not sure if `room_id` or `user_id` can be `None` here, after looking at `check_user_in_room_or_world_readable`'s docstring:

https://github.com/matrix-org/synapse/blob/38e1fac8861f12b707609da06008695a05aaf21c/synapse/api/auth.py#L638-L655

Feel free to look at/fix in a separate issue/PR though.

Reply:
Yeah, looks incorrect to me too. I'll just fix it here, assuming mypy passes.

Reply:
(There's also only 1 caller, which provides both parameters so...)